x/ccv/provider/keeper/grpc_query.go

```diff
@@ -232,18 +229,40 @@ func (k Keeper) QueryRegisteredConsumerRewardDenoms(goCtx context.Context, req *
 	}, nil
 }
 
-// TODO (PERMISSIONLESS)
 func (k Keeper) QueryProposedConsumerChainIDs(goCtx context.Context, req *types.QueryProposedChainIDsRequest) (*types.QueryProposedChainIDsResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	chains := k.GetAllProposedConsumerChainIDs(ctx)
+	var proposals []govv1.Proposal
+	err := k.govKeeper.Proposals.Walk(ctx, nil, func(proposalId uint64, proposal govv1.Proposal) (stop bool, err error) {
+		proposals = append(proposals, proposal)
+		return false, nil
+	})
+	if err != nil {
+		return nil, status.Error(codes.Internal, "failed to retrieve proposals")
+	}
+
+	var proposedChains []types.ProposedChain
+	for _, proposal := range proposals {
+		status := proposal.Status
+		if status != govv1.ProposalStatus_PROPOSAL_STATUS_DEPOSIT_PERIOD && status != govv1.ProposalStatus_PROPOSAL_STATUS_VOTING_PERIOD {
+			// we only care about active proposals
+			continue
+		}
+
+		messages := proposal.GetMessages()
+		for _, msg := range messages {
+			if sdkMsg, isMsgUpdateConsumer := msg.GetCachedValue().(*types.MsgUpdateConsumer); isMsgUpdateConsumer {
+				proposedChains = append(proposedChains, types.ProposedChain{ProposalID: proposal.Id, ConsumerId: sdkMsg.ConsumerId})
```

Comment:
Why is the chainId in the ProposedChain deprecated?

Reply:
Removed this query; see comment [here](https://github.com/cosmos/interchain-security/pull/2177/files#r1734448221).